src/main/java/bg/codexio/springframework/data/jpa/requery/payload/FilterRequestWrapper.java

```diff
@@ -0,0 +1,59 @@
+package bg.codexio.springframework.data.jpa.requery.payload;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+public class FilterRequestWrapper {
+    private List<FilterRequest> filterRequest;
+    private FilterGroupRequest filterGroupRequest;
+
+    public FilterRequestWrapper() {
+    }
+
+    public FilterRequestWrapper(List<FilterRequest> filterRequest) {
+        setFilterRequest(filterRequest);
+    }
+
+    public FilterRequestWrapper(FilterGroupRequest filterGroupRequest) {
+        setFilterGroupRequest(filterGroupRequest);
+    }
+
+    public List<FilterRequest> getFilterRequest() {
+        return filterRequest;
+    }
+
+    public void setFilterRequest(List<FilterRequest> filterRequest) {
```

Comment:
Does this need public setters? I would treat the wrapper as an immutable object. It can be even a record?

Reply:
Records can have overloaded constructors as well: https://docs.oracle.com/en/java/javase/22/language/records.html (see chapter `Alternative Record Constructors`)

Reply:
Refactored to Record with Optional fields and overloaded constructors for handling different instantiation.